### PR TITLE
287 update identity to v0.6

### PR DESCRIPTION
**Pull request by dominic22**
In order to run the ssi-bridge you need to create a new root-identity and delete the old root-identity.

1. Delete old root-identity
2. npm install
3. npm run setup-api
4. npm run start

### CODE REVIEW

**Review by JuriBog — APPROVED** [review]

File: api/ssi-bridge/src/test/mocks/bitmap.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Bitmap } from '@iota/is-shared-modules/lib/web/models/types/identity';
```
Comment (Schereo) [inline api/ssi-bridge/src/test/mocks/bitmap.ts:1]: With the new shared-modules a simplified import should be possible:
```suggestion
import { Bitmap } from '@iota/is-shared-modules';
```

File: api/ssi-bridge/src/test/mocks/identities.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import { Encoding, IdentityJson, IdentityKeys, User, UserType } from '@iota/is-shared-modules';
+import { IdentityKeys, User } from '@iota/is-shared-modules';
+import { IdentityDocument, Keys } from '@iota/is-shared-modules';
```
Comment (Schereo) [inline api/ssi-bridge/src/test/mocks/identities.ts:2]: Why not import both in the same line?

File: api/ssi-bridge/src/tools/migration/migration-to-0.1.3.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import * as dotenv from 'dotenv';
-dotenv.config();
 import { MongoDbService } from '@iota/is-shared-modules/node';
 
+dotenv.config();
```
Comment (Schereo) [inline api/ssi-bridge/src/tools/migration/migration-to-0.1.3.ts:4]: Why was this moved?

File: api/ssi-bridge/src/tools/open-api-creator/index.ts [hunk 4]
```diff
@@ -110,7 +112,7 @@ const converter = () => {
 		if (err) {
 			console.log(err);
 		} else {
-			console.log('Successfully converted JSON schema.');
+			console.log('Successfully converted  schema.');
```
Comment (Schereo) [inline api/ssi-bridge/src/tools/open-api-creator/index.ts:115]:
```suggestion
			console.log('Successfully converted schema.');
```

File: api/shared-modules/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@iota/is-shared-modules",
-  "version": "0.1.20",
+  "version": "0.2.0-alpha.7",
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/shared-modules/package.json:3]: since there might be some changes, need to check with michele

File: api/ssi-bridge/src/services/verification-service.ts [hunk 6]
```diff
@@ -30,27 +29,27 @@ export class VerificationService {
...
-	async getKeyCollection(keyCollectionIndex: number) {
-		let keyCollection = await KeyCollectionDb.getKeyCollection(keyCollectionIndex, this.configService.serverIdentityId, this.serverSecret);
-		if (!keyCollection) {
-			keyCollection = await this.generateKeyCollection(keyCollectionIndex, this.keyCollectionSize, this.configService.serverIdentityId);
-			const res = await KeyCollectionDb.saveKeyCollection(keyCollection, this.configService.serverIdentityId, this.serverSecret);
+	async getBitmap(bitmapIndex: number) {
+		let bitmap = await BitmapDb.getBitmap(bitmapIndex, this.configService.serverIdentityId);
+		if (!bitmap) {
+			bitmap = await this.createRevocationBitmap(bitmapIndex, this.configService.serverIdentityId);
+			console.log('saving bitmap....');
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/services/verification-service.ts:38]: no, thanks

File: api/ssi-bridge/src/routes/authentication/index.test.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { UserService } from '../../services/user-service';
 import { StatusCodes } from 'http-status-codes';
 import * as AuthDb from '../../database/auth';
 import { User } from '@iota/is-shared-modules';
-import * as EncryptionUtils from '@iota/is-shared-modules/node/utils/encryption';
+import * as EncryptionUtils from '@iota/is-shared-modules/node/utils/encryption'; // TODO this import won't work with @iota/is-shared-modules/node
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/routes/authentication/index.test.ts:7]: true now it works

File: api/ssi-bridge/src/routers/identity/index.ts [hunk 8]
```diff
@@ -191,31 +191,6 @@ identityRouter.get('/identity/:id', apiKeyMiddleware, authMiddleWare, getUser);
  *         application/json:
  *           schema:
  *             $ref: "#/components/schemas/IdentitySchema"
- *           example:
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/routers/identity/index.ts:194]: maybe we'll add some in a sperate task

File: api/ssi-bridge/src/tools/migration/migration-to-0.1.3.ts (hunk 3 above)
Comment (dominic22) [reply api/ssi-bridge/src/tools/migration/migration-to-0.1.3.ts:4]: reverted

File: api/ssi-bridge/src/routes/verification/tests/revoke-vc.test.ts [hunk 9]
```diff
@@ -362,44 +354,6 @@ describe('test authentication routes', () => {
 			expect(loggerSpy).not.toHaveBeenCalled();
 			expect(nextMock).not.toHaveBeenCalled();
 		});
-
-		it('identity is already revoked', async () => {
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/routes/verification/tests/revoke-vc.test.ts:366]: The implementation has changed. Before we were able to check this. Now we don't get this information anymore since `revokeCredentials` does not provide if the credential was already revoked or not anymore on sdk side

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 10]
```diff
@@ -84,164 +69,175 @@ export class SsiService {
...
 	async createVerifiableCredential<T>(
-		issuerIdentity: IdentityKeys,
+		identityKeys: IdentityKeys,
 		credential: Credential<T>,
-		keyCollectionJson: KeyCollectionJson,
-		keyCollectionIndex: number,
+		bitmapIndex: number,
 		subjectKeyIndex: number
-	): Promise<VerifiableCredentialJson> {
+	): Promise<any> {
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/services/ssi-service.ts:76]: there is, added it

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 11]
```diff
@@ -84,164 +69,175 @@ export class SsiService {
...
 			throw new Error('could not parse key or doc of the identity');
 		}
 	}
 
-	generateIdentity() {
+	async generateIdentity(): Promise<{ account: Identity.Account; doc: Identity.Document; key: Identity.KeyPair }> {
 		try {
-			const { doc, key } = new Document(this.config.keyType, Identity.Network.mainnet().toString()) as any;
+			const builder = this.getAccountBuilder();
+			const keyPair = new KeyPair(KeyType.Ed25519);
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/services/ssi-service.ts:207]: this is the signing key so it will stay the same. we will also create a encryption key

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 12]
```diff
@@ -84,164 +69,175 @@ export class SsiService {
...
-	private addPropertyToDoc(doc: Identity.Document, property: { [key: string]: any }): Identity.Document {
-		return Identity.Document.fromJSON({
-			...doc.toJSON(),
-			...property
-		});
+	private getConfig(usePermaNode?: boolean): Identity.IClientConfig {
+		return {
+			permanodes: usePermaNode ? [{ url: this.config.permaNode }] : [],
+			primaryNode: { url: this.config.node },
+			network: Identity.Network.mainnet(),
```
[the remark this answers is not in the document]
Comment (dominic22) [reply api/ssi-bridge/src/services/ssi-service.ts:239]: I will take a note how to handle this properly... testnet can be used by adjusting the nodes i think this will mainly affect if the did starts with did:iota:.. or did:iota-test:...
